src/callbacks_step/analysis_surface_2d.jl

```diff
@@ -0,0 +1,148 @@
+# This file contains callbacks that are performed on the surface like computation of
+# surface forces
+
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+# The boundary_index is chosen so that
+# semi.boundary_conditions.boundary_indices[boundary_index]
+# gives the solution point indices on which the function `variable` will compute
+# the quantity of interest. The `variable` can, e.g., integrate over all indices
+# to compute the coefficient of lift. But it can also be used to loop over all indices
+# to save coefficient of pressure versus surface points in a data file
+# The struct contains an inner constructor which helps the user choose those indices
+# by specifying the `boundary_condition_type` of the indices. The user needs to make
+# sure that they choose the `boundary_condition_type` so that it is only applying to the
+# parts of boundary that are of interest
```

Comment:
If we stick with the proposed approach this needs to be adjusted accordingly. Also we should add a docstring.